### PR TITLE
Improve swagger models

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

This PR consists:
- fix api order to previus (like in [demo](demo.sonata-project.org/api/doc)) - `pedantic`
- change swagger models to selialized models - this will allow see correct return example in documentation (under `/api/doc`)  
image

- clean up docblocks - remove unnesessery commets
- improve doc

I am targeting this branch, because this change respect BC.

### CODE REVIEW

File: docs/reference/advanced_configuration.rst [hunk 1]
```diff
@@ -59,9 +59,8 @@ Full configuration options:
     jms_serializer:
         metadata:
             directories:
-                App:
-                    path: "%kernel.root_dir%/../vendor/sonata-project/user-bundle/Sonata/UserBundle/Resources/config/serializer/FOSUserBundle"
-                    namespace_prefix: 'FOS\UserBundle'
+                - { name: 'sonata_user', path: "%kernel.root_dir%/../vendor/sonata-project/user-bundle/src/Resources/config/serializer", namespace_prefix: 'Sonata\UserBundle' }
+                - { name: 'fos_user', path: "%kernel.root_dir%/../vendor/sonata-project/user-bundle/src/Resources/config/serializer/FOSUserBundle", namespace_prefix: 'FOS\UserBundle' }
```
Comment (phansys) [inline docs/reference/advanced_configuration.rst:63]:
```suggestion
                - { name: 'sonata_user', path: "%kernel.project_dir%/vendor/sonata-project/user-bundle/src/Resources/config/serializer", namespace_prefix: 'Sonata\UserBundle' }
                - { name: 'fos_user', path: "%kernel.project_dir%/vendor/sonata-project/user-bundle/src/Resources/config/serializer/FOSUserBundle", namespace_prefix: 'FOS\UserBundle' }
```

File: docs/reference/api.rst [hunk 2]
```diff
@@ -35,6 +35,12 @@ Here's the configuration we used, you may adapt it to your needs:
         router: { annotations: true }
         request: { converters: true }
 
+    jms_serializer:
+        metadata:
+            directories:
+                - { name: 'sonata_user', path: "%kernel.root_dir%/../vendor/sonata-project/user-bundle/src/Resources/config/serializer", namespace_prefix: 'Sonata\UserBundle' }
+                - { name: 'fos_user', path: "%kernel.root_dir%/../vendor/sonata-project/user-bundle/src/Resources/config/serializer/FOSUserBundle", namespace_prefix: 'FOS\UserBundle' }
```
Comment (phansys) [inline docs/reference/api.rst:42]:
```suggestion
                - { name: 'sonata_user', path: "%kernel.project_dir%/vendor/sonata-project/user-bundle/src/Resources/config/serializer", namespace_prefix: 'Sonata\UserBundle' }
                - { name: 'fos_user', path: "%kernel.project_dir%/vendor/sonata-project/user-bundle/src/Resources/config/serializer/FOSUserBundle", namespace_prefix: 'FOS\UserBundle' }
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Changelog is needed, isn't it ?

**Comment by wbloszyk** [comment]
It is not, becouse we don't release support for it.

**Comment by phansys** [comment]
Could you please explain how these changes are related to the used serializer? Or in this case, this change is only about documentation?

**Comment by wbloszyk** [comment]
> Could you please explain how these changes are related to the used serializer? Or in this case, this change is only about documentation?

This change is only in documentation (doc path) and in swagger documentation (under `/api/doc` path). People get the same result from API like before this PR.

Information about this directories should be mark, becouse:
- 'fos_user' directory provide only fos user fields
- 'sonata_user' directory provide only sonata user fields
- using both directory will return all fields for user

**Comment by wbloszyk** [comment]
@greg0ire @phansys https://github.com/sonata-project/SonataUserBundle/pull/1225#discussion_r493012141
Any idea how to improve it?

**Comment by phansys** [comment]
This can be improved, but as a first step, I think the description should be updated this way:

* {Change|Reverted} order of the defined ReST API paths in the routing definition files (`user.xml`, `group.xml`) in order to respect the order they had previous to {version|pr|commit};
* {Change|Reverted} the class names passed as `type` attribute for the `@Nelmio\ApiDocBundle\Annotation\Model` annotations used as part of the Swagger definition in the controllers under `Sonata\UserBundle\Controller\Api\` namespace;
* Updated docs in order to expose how to configure custom serialization paths under `jms_serializer` configuration node (`jms_serializer.metadata.directories`).

Also, I think the description should provide a clear idea why these changes are required (if they are bugfixes) or how they can improve the current codebase.

**Comment by wbloszyk** [comment]
IMO changelog for docs(with jms_serializer info) will be enought. Change order is only `pedantic`. We cant talk about revert/change type becouse it is just added. (it is not in nelmio v2).

**Comment by phansys** [comment]
My suggestion is not for the changelog section. I want to understand the motivation of these changes and how they are intended to work.

**Comment by wbloszyk** [comment]
> My suggestion is not for the changelog section. I want to understand the motivation of these changes and how they are intended to work.

Oh, we have Serialization section (in api.rst) for it.

**Comment by phansys** [comment]
Sorry if I'm not clear enough.

>Also, I think the description should provide a clear idea why these changes are required (if they are bugfixes) or how they can improve the current codebase.

>I want to understand the motivation of these changes and how they are intended to work.

I'm asking you to explain (in the PR description) why do you think these changes are required (what they fix or what they improve).

**Comment by wbloszyk** [comment]
> I'm asking you to explain (in the PR description) why do you think these changes are required (what they fix or what they improve).

Updated.

**Comment by phansys** [comment]
I think we should clarify some assumptions:
* The [Open API Specification](https://swagger.io/specification/) (formerly Swagger) defines an API itself;
* The Open API Specification generated by the annotations in our controllers is used internally to generate the presentation layer  (and the sandbox) for our ReST API; but that is not the only consumer able to use that definition.
 The resulting definition can be used by third party applications to generate HTTP clients, SDKs or even other sandbox or documentation renderings (by instance, you should be able to render the generated spec in https://editor.swagger.io/, also, you can create a SDK using https://swagger.io/tools/swagger-codegen/; you could import that spec locally to be used with [Postman](https://learning.postman.com/docs/getting-started/importing-and-exporting-data/#importing-api-specifications) too).

Given that, I think we should apply our BC promise to that spec and try to avoid breaking changes. So, if we make some change on the OAS we are exposing, I think we should use the same reasoning that we use for the rest of our API.

Please, let me know if you have some concerns about this.

**Comment by wbloszyk** [comment]
We (becouse I also based on your work) did not chane any result which will be the same after upgrade `nelmio`. Result is generated by `FOSRestBundle` and sonata controllers. 

In `nelmio` v2 documentation is generated on own code.

In `nelmio` v3 documentation is generated based on swagger ([zircote/swagger-php](https://github.com/zircote/swagger-php)). 
In this case we return the same results but integrated with Swagger UI). We will have other, better documentation page with model example. I don't know about is current API in OpenApi standard but for sure it should be BC. I can compare API from old and new system to be sure in 1000%.

**Comment by phansys** [comment]
I think these facts are not compatible with each other:

>change swagger models to selialized models

>In this case we return the same results but integrated with Swagger UI

If the serialization was changed, the resulting models will be different (unless the serialization is not adding any value).

**Comment by wbloszyk** [comment]
> I think these facts are not compatible with each other:
> 
> > change swagger models to selialized models
```php
from: @SWG\Schema(ref=@Model(type="Sonata\UserBundle\Model\UserInterface"))
to: @SWG\Schema(ref=@Model(type=User::class, groups={"sonata_api_read"})) //Sonata\UserBundle\Model\User
```
This will allow intagrate jms_serializer.metadata.directories with Swagger UI. Like you see `UserInterface` is empty.

image

> > In this case we return the same results but integrated with Swagger UI
> 
> If the serialization was changed, the resulting models will be different (unless the serialization is not adding any value).

Change will be when people change it in own configuration. Also like I said, we use FOSRestBundle to return API. `/api/user/user/1.json` will return the same value with `nelmio` v3 or without.

**Comment by phansys** [comment]
>Form Swagger (open API) we have only UI.

I think this is a wrong assumption.
I tried to explain that the UI rendering is not the only purpose for the spec in my previous comment https://github.com/sonata-project/SonataUserBundle/pull/1237#issuecomment-699057873.

Please, register the following route in your environment:
https://github.com/sonata-project/SonataUserBundle/blob/9fee1fabac619e763878be36afa0e7c4ca5d5ee4/tests/App/Resources/config/routing/api_nelmio_v3.yml#L1-L4
then, get the HTTP response from `/api/doc.json` path.
At this point, you should be able to see the OAS definition in JSON.
That's the actual interface (not UI, but programmatic) provided by OAS.
IMO, the comparison must be done based on that output.

**Comment by wbloszyk** [comment]
What diffrence with compare Swagger UI with `/api/doc.json` when `/api/doc.json` is used to generate API?

Anyway I can do it if you want:
```json
{"swagger":"2.0","info":{"title":"Sonata Sandbox","description":"This is demo RestFul API!","version":"1.0.0"},"paths":{"\/api\/user\/groups":{"get":{"summary":"Returns a paginated list of groups.","parameters":[{"name":"page","in":"query","allowEmptyValue":false,"required":false,"description":"Page for groups list pagination (1-indexed)","type":"string","default":"1"},{"name":"count","in":"query","allowEmptyValue":false,"required":false,"description":"Number of groups per page","type":"string","default":"10"},{"name":"orderBy","in":"query","required":false,"description":"Query groups order by clause (key is field, value is direction)","type":"string"},{"name":"enabled","in":"query","allowEmptyValue":true,"required":false,"description":"Enables or disables the groups only?","type":"string"},{"name":"orderBy[]","in":"query","allowEmptyValue":true,"required":false,"description":"Query groups order by clause (key is field, value is direction)","items":{"type":"string","pattern":"ASC|DESC"},"type":"array","collectionFormat":"multi"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/BasePager"}}},"tags":["\/api\/user\/groups"]},"post":{"summary":"Adds a group.","responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/Group"}},"400":{"description":"Returned when an error has occurred during the group creation"}},"tags":["\/api\/user\/groups"]}},"\/api\/user\/groups\/{id}":{"get":{"summary":"Retrieves a specific group.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/Group"}},"404":{"description":"Returned when group is not found"}},"tags":["\/api\/user\/groups"]},"put":{"summary":"Updates a group.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/Group"}},"400":{"description":"Returned when an error has occurred during the group creation"},"404":{"description":"Returned when unable to find group"}},"tags":["\/api\/user\/groups"]},"delete":{"summary":"Deletes a group.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when group is successfully deleted"},"400":{"description":"Returned when an error has occurred during the group deletion"},"404":{"description":"Returned when unable to find group"}},"tags":["\/api\/user\/groups"]}},"\/api\/user\/users":{"get":{"summary":"Returns a paginated list of users.","parameters":[{"name":"page","in":"query","allowEmptyValue":false,"required":false,"description":"Page for users list pagination (1-indexed)","type":"string","default":"1"},{"name":"count","in":"query","allowEmptyValue":false,"required":false,"description":"Number of users per page","type":"string","default":"10"},{"name":"orderBy","in":"query","required":false,"description":"Query users order by clause (key is field, value is direction)","type":"string"},{"name":"enabled","in":"query","allowEmptyValue":true,"required":false,"description":"Enables or disables the users only?","type":"string"},{"name":"orderBy[]","in":"query","allowEmptyValue":true,"required":false,"description":"Query users order by clause (key is field, value is direction)","items":{"type":"string","pattern":"ASC|DESC"},"type":"array","collectionFormat":"multi"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/BasePager"}}},"tags":["\/api\/user\/users"]},"post":{"summary":"Updates a user.","responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/User"}},"400":{"description":"Returned when an error has occurred during the user creation"},"404":{"description":"Returned when unable to find user"}},"tags":["\/api\/user\/users"]}},"\/api\/user\/users\/{id}":{"get":{"summary":"Retrieves a specific user.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/User"}},"404":{"description":"Returned when user is not found"}},"tags":["\/api\/user\/users"]},"put":{"summary":"Updates a user.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/User"}},"400":{"description":"Returned when an error has occurred during the user creation"},"404":{"description":"Returned when unable to find user"}},"tags":["\/api\/user\/users"]},"delete":{"summary":"Deletes an user.","parameters":[{"name":"id","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when user is successfully deleted"},"400":{"description":"Returned when an error has occurred during the user deletion"},"404":{"description":"Returned when unable to find user"}},"tags":["\/api\/user\/users"]}},"\/api\/user\/users\/{userId}\/groups\/{groupId}":{"post":{"summary":"Attach a group to a user.","parameters":[{"name":"userId","in":"path","required":true,"type":"string"},{"name":"groupId","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/User"}},"400":{"description":"Returned when an error has occurred while user\/group attachment"},"404":{"description":"Returned when unable to find user or group"}},"tags":["\/api\/user\/users"]},"delete":{"summary":"Detach a group to a user.","parameters":[{"name":"userId","in":"path","required":true,"type":"string"},{"name":"groupId","in":"path","required":true,"type":"string"}],"responses":{"200":{"description":"Returned when successful","schema":{"$ref":"#\/definitions\/User"}},"400":{"description":"Returned when an error occurred while detaching the user from the group"},"404":{"description":"Returned when unable to find user or group"}},"tags":["\/api\/user\/users"]}},"\/api\/doc.json":{"get":{"responses":{"default":{"description":""}}}}},"definitions":{"BasePager":{"properties":{"page":{"type":"integer"},"per_page":{"type":"integer"},"last_page":{"type":"integer"},"total":{"type":"integer"}},"type":"object"},"Group":{"properties":{"name":{"type":"string"}},"type":"object"},"User":{"properties":{"username":{"type":"string"},"username_canonical":{"type":"string"},"email":{"type":"string"},"email_canonical":{"type":"string"},"enabled":{"type":"boolean"},"last_login":{"type":"string","format":"date-time"},"confirmation_token":{"title":"Random string sent to the user email address in order to verify it.","type":"string"},"password_requested_at":{"type":"string","format":"date-time"},"roles":{"additionalProperties":true,"type":"object"},"facebook_data":{"title":"{@inheritdoc}","additionalProperties":true,"type":"object"},"twitter_data":{"title":"{@inheritdoc}","additionalProperties":true,"type":"object"},"gplus_data":{"title":"{@inheritdoc}","additionalProperties":true,"type":"object"},"created_at":{"type":"string","format":"date-time"},"updated_at":{"type":"string","format":"date-time"},"two_step_verification_code":{"type":"string"},"date_of_birth":{"type":"string","format":"date-time"},"firstname":{"type":"string"},"lastname":{"type":"string"},"website":{"type":"string"},"biography":{"type":"string"},"gender":{"type":"string"},"locale":{"type":"string"},"timezone":{"type":"string"},"phone":{"type":"string"},"facebook_uid":{"type":"string"},"facebook_name":{"type":"string"},"twitter_uid":{"type":"string"},"twitter_name":{"type":"string"},"gplus_uid":{"type":"string"},"gplus_name":{"type":"string"},"token":{"type":"string"}},"type":"object"}},"host":"127.0.0.1"}
```

**Comment by phansys** [comment]
>What diffrence with compare Swagger UI with /api/doc.json when /api/doc.json is used to generate API?

IMO, the important thing is not the resulting UI, since it's just a graphic implementation of the spec.

>With nelmio v3 we will use Swagger UI based on OAS definition in JSON. We will get the same API like in nelmio v2 + additional this definition.

Regardless the version of `nelmio/api-doc-bundle`, you can always get the OAS spec you have defined. By instance, in version 2.x this is the [action](https://github.com/nelmio/NelmioApiDocBundle/blob/28802f2c44dbbf29aa7f5dc80a10f44d3558f580/Controller/ApiDocController.php#L31) responsible for that. And even, if there isn't any associated action, the package is always aware of the specification you  have defined.

What I'm trying to explain is, if we change that definition in some way that it becomes incompatible with a previous version in a *MINOR* or *PATCH* release, we should make sure there is a good reason for that. In that case, we should provide clear and specific information in order to tell why these changes are required.

Let's use the following sample scenario:
* You are providing a ReST API for third party packages, like a SPA or a mobile app;
* These third party packages are using the OAS spec you're providing to generate its own API implementations for the language they use (a custom HTTP client / SDK);
* You change some structure in your spec that breaks the compatibility with a previous version (by removing a response property or changing its type, adding a new requirement, etc);
* The HTTP clients / SDKs built previous to your changes will stop working, since now your API has a different contract that it had when these pieces of software were using before.

Please, let me know if you have any doubts about this.

**Comment by wbloszyk** [comment]
### Nelmio versions corelated with Swagger/OpenApi
- `nelmio/api-doc-bundle` v2 => swagger v1
- `nelmio/api-doc-bundle` v3 => swagger v2
- `nelmio/api-doc-bundle` v4 => OpenApi v3 (**Final** version **was released** ~20h ago)

### now
- we used `nelmio/api-doc-bundle` v2
- we can return swagger OAS spec v1 (probably)
- OAS (swaggerVersion 1.x) will return wrong model - example "Sonata\UserBundle\Model\UserInterface"
    - should "App\Entity\User" then model will return correct metadata
- to return correct data people must add serializer for own Entity(to return own fields like [here](https://github.com/sonata-project/sandbox-build/blob/2.3/src/Application/Sonata/UserBundle/Resources/config/serializer/Entity.User.xml))
    - otherwise some data like `id` will be missing

### we adding
- add support for `nelmio/api-doc-bundle` v3 
- allow return swagger OAS spec v2
- we should provide (by recipies and doc) serializator and models for App\Entity\*

### conclusion
- return data will be serialize to the same results
- OAS will be diffrence (current OAS is wrong)

### support for `nelmio/api-doc-bundle` v4
IMO We can add support for it becouse it is in stable release. WDYT? @sonata-project/contributors 

@phansys 
If you agree I will be work on it tomorrow. IMO we should do it becouse return OAS models with empty data is unhelpful.

Here is proof:
```json
{"swaggerVersion":"1.2","apiVersion":"0.1","basePath":"\/api","resourcePath":"\/user-users","apis":[{"path":"\/user\/users.{_format}","operations":[{"method":"GET","summary":"Returns a paginated list of users.","nickname":"get_user-users","parameters":[{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]},{"paramType":"query","name":"page","type":"string","description":"Page for users list pagination (1-indexed)"},{"paramType":"query","name":"count","type":"string","description":"Number of users by page"},{"paramType":"query","name":"orderBy","type":"string","description":"Query users order by clause (key is field, value is direction"},{"paramType":"query","name":"enabled","type":"string","description":"Enabled\/disabled users only?"}],"responseMessages":[{"code":200,"message":"See standard HTTP status code reason for 200","responseModel":"Sonata.DatagridBundle.Pager.PagerInterface"}],"type":"Sonata.DatagridBundle.Pager.PagerInterface"},{"method":"POST","summary":"Adds an user.","nickname":"post_user-users","parameters":[{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when successful","responseModel":"Sonata.UserBundle.Model.User"},{"code":400,"message":"Returned when an error has occurred while user creation"}],"type":"Sonata.UserBundle.Model.User"}]},{"path":"\/user\/users\/{id}.{_format}","operations":[{"method":"DELETE","summary":"Deletes an user.","nickname":"delete_user-users","parameters":[{"paramType":"path","name":"id","type":"string","required":true},{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when user is successfully deleted"},{"code":400,"message":"Returned when an error has occurred while user deletion"},{"code":404,"message":"Returned when unable to find user"}]},{"method":"GET","summary":"Retrieves a specific user.","nickname":"get_user-users","parameters":[{"paramType":"path","name":"id","type":"string","required":true},{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when successful","responseModel":"Sonata.UserBundle.Model.UserInterface"},{"code":404,"message":"Returned when user is not found"}],"type":"Sonata.UserBundle.Model.UserInterface"},{"method":"PUT","summary":"Updates an user.","nickname":"put_user-users","parameters":[{"paramType":"path","name":"id","type":"string","required":true},{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when successful","responseModel":"Sonata.UserBundle.Model.User"},{"code":400,"message":"Returned when an error has occurred while user creation"},{"code":404,"message":"Returned when unable to find user"}],"type":"Sonata.UserBundle.Model.User"}]},{"path":"\/user\/users\/{userId}\/groups\/{groupId}.{_format}","operations":[{"method":"DELETE","summary":"Detach a group to a user.","nickname":"delete_user-users","parameters":[{"paramType":"path","name":"userId","type":"string","required":true},{"paramType":"path","name":"groupId","type":"string","required":true},{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when successful","responseModel":"Sonata.UserBundle.Model.User"},{"code":400,"message":"Returned when an error has occurred while user\/group detachment"},{"code":404,"message":"Returned when unable to find user or group"}],"type":"Sonata.UserBundle.Model.User"},{"method":"POST","summary":"Attach a group to a user.","nickname":"post_user-users","parameters":[{"paramType":"path","name":"userId","type":"string","required":true},{"paramType":"path","name":"groupId","type":"string","required":true},{"paramType":"path","name":"_format","type":"string","required":true,"enum":["json","xml","html"]}],"responseMessages":[{"code":200,"message":"Returned when successful","responseModel":"Sonata.UserBundle.Model.User"},{"code":400,"message":"Returned when an error has occurred while user\/group attachment"},{"code":404,"message":"Returned when unable to find user or group"}],"type":"Sonata.UserBundle.Model.User"}]}],"models":{"Sonata.DatagridBundle.Pager.PagerInterface":{"id":"Sonata.DatagridBundle.Pager.PagerInterface","description":"","properties":[],"required":[]},"Sonata.UserBundle.Model.User":{"id":"Sonata.UserBundle.Model.User","description":"","properties":{"username":{"type":"string","description":null},"email":{"type":"string","description":null},"plainPassword":{"type":"string","description":null},"facebook_data":{"type":"array","description":"{@inheritdoc}","items":{"type":"string"}},"twitter_data":{"type":"array","description":"{@inheritdoc}","items":{"type":"string"}},"gplus_data":{"type":"array","description":"{@inheritdoc}","items":{"type":"string"}},"created_at":{"type":"string","description":"DateTime","format":"date-time"},"updated_at":{"type":"string","description":"DateTime","format":"date-time"},"two_step_verification_code":{"type":"string","description":"string"},"date_of_birth":{"type":"string","description":"DateTime","format":"date-time"},"firstname":{"type":"string","description":"string"},"lastname":{"type":"string","description":"string"},"website":{"type":"string","description":"string"},"biography":{"type":"string","description":"string"},"gender":{"type":"string","description":"string"},"locale":{"type":"string","description":"string"},"timezone":{"type":"string","description":"string"},"phone":{"type":"string","description":"string"},"facebook_uid":{"type":"string","description":"string"},"facebook_name":{"type":"string","description":"string"},"twitter_uid":{"type":"string","description":"string"},"twitter_name":{"type":"string","description":"string"},"gplus_uid":{"type":"string","description":"string"},"gplus_name":{"type":"string","description":"string"},"token":{"type":"string","description":"string"}},"required":["username","email","plainPassword"]},"Sonata.UserBundle.Model.UserInterface":{"id":"Sonata.UserBundle.Model.UserInterface","description":"","properties":[],"required":[]}},"produces":[],"consumes":[],"authorizations":[]}
```

**Comment by phansys** [comment]
If we are returning empty data in a model which should be exposing data, I expect the PR description to be explicit about what models were involved in this issue.
Even, since I'm aware there are also changes in the attributes passed to the `@Nelmio\ApiDocBundle\Annotation\Model` annotation that modify the documentation for the endpoints it affects, I think the current description lacks the information required by other contributors to decide if these changes are fine given its purpose or nature (pedantic, patch, feature, major).

IMO, regardless the version of the supported integration with `nelmio/api-doc-bundle`, you should invest a little more effort in making a clear description, focusing in the definition instead of what is rendered on the demo page or the sandbox.
By instance, when you change a method signature in a class, you explain what is changed (if an argument has changed its type, if a missing default value was added, if the return type is updated, etc). I think the same kind of explanation is required here.

Last, I'd suggest waiting for the revision by other contributors, since it seems that I'm the only one involved in this review by the moment.

**Comment by wbloszyk** [comment]
Close in favor for #1239

**Pull request closed without merging**